### PR TITLE
Move olx.render.* and related typedefs to ol/render/*

### PR DESCRIPTION
**Pull request by pfirpfel**
Changes according to #7838, subtask of #7947

externs/olx.js is basically empty now. I did not remove it, because this would require additional changes which are out of scope for my task.

### CODE REVIEW

**Comment by pfirpfel** [comment]
Rebased with master to resolve conflicts

File: src/ol/render/canvas.js [hunk 1]
```diff
@@ -8,6 +8,66 @@ import LRUCache from '../structs/LRUCache.js';
...
+/**
+ * @typedef {{fillStyle: module:ol/colorlike~ColorLike}} FillState
+ */
+
+
+/**
+ * @typedef {Object} FillStrokeState
+ * @property {module:ol/colorlike~ColorLike} [currentFillStyle]
+ * @property {module:ol/colorlike~ColorLike} [currentStrokeStyle]
+ * @property string{} [currentLineCap]
```
Comment (tschaub) [inline src/ol/render/canvas.js]: This should be `{string}` instead of `string{}`.

Comment (pfirpfel) [reply src/ol/render/canvas.js]: Thanks

**Comment by pfirpfel** [comment]
Rebased with master to fix conflicts

**Comment by pfirpfel** [comment]
`ol/render~State` still had an error. I put the fix into the very first commit. Tests are passing now.

**Comment by tschaub** [comment]
Thanks for the work on this @pfirpfel.

File: src/ol/render/canvas.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ import {create as createTransform} from '../transform.js';
 
 
 /**
- * @typedef {{fillStyle: module:ol/colorlike~ColorLike}} FillState
+ * @typedef {Object} FillState
+ * @property {module:ol/colorlike~ColorLike} fillStyle
```
[the remark this answers is not in the document]
Comment (fredj) [reply src/ol/render/canvas.js]: yes, it was causing babel to crash (out of memory)

**Pull request merged**